test/e2e/app-dir/segment-cache/incremental-opt-in/segment-cache-incremental-opt-in.test.ts

```diff
@@ -21,21 +21,22 @@ describe('segment cache (incremental opt in)', () => {
     // that occur. Then at the end we confirm there are no duplicates.
     const prefetches = new Map()
     const duplicatePrefetches = new Map()
+    const unexpectedResponse = []
 
-    let act
+    let currentPage: Page
     const browser = await next.browser('/', {
       async beforePageLoad(page) {
-        act = createRouterAct(page)
+        currentPage = page
         await page.route('**/*', async (route) => {
           const request = route.request()
+          const headers = await request.allHeaders()
           const isPrefetch =
-            request.headerValue('rsc') !== null &&
-            request.headerValue('next-router-prefetch') !== null
```

Comment:
Those were unresolved promises, so `isPrefetch` was always true and registered any request, e.g. for js assets.